rafs/src/metadata/mod.rs

```diff
@@ -744,27 +744,37 @@ impl RafsSuper {
         Ok(())
     }
 
+    /// Walkthrough inode tree by DFS order, please ensure ino is not a hardlink,
+    /// otherwise will lose some hardlink pairs.
     pub fn walk_inodes(
```

Comment:
How do we ensure `ino` is not a hardlink? I don't see anywhere we put such enforcement.

Reply:
There is no such enforcement, we usually call this method for the directory, and it is currently only used for root ino.

Reply:
IMO,
1. If we only use it for root ino (no plan for other usage), let's expose a `walk_root_tree`-like method so that it will not be called in other ways.
2. If we plan to use it for other inodes other than root, we need to add a check in `walk_inodes` to make sure the passed in inode is not a hardlink.

Reply:
Agree with point 2, but I think it makes no sense to walk a non-directory for now, how about checking that ino is a directory first? Also change the func name to `walk_dir`. This is easier than checking if an ino has a hardlink pair.

Reply:
Fixed